src/Microsoft.ML.AutoMLFeaturizers/ToStringTransformer.cs

```diff
@@ -0,0 +1,1763 @@
+        }
+
+        private protected override void SaveModel(ModelSaveContext ctx)
+        {
+            Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            // *** Binary format ***
+            // int number of column pairs
+            // for each column pair:
+            //      string output column  name
+            //      string input column name
+            //      string representation of type
+            //      int c++ state array length
+
+            ctx.Writer.Write(_columns.Count());
+            foreach (var column in _columns)
+            {
+                ctx.Writer.Write(column.Name);
+                ctx.Writer.Write(column.Source);
+                ctx.Writer.Write(column.Type);
+
+                var data = column.CreateTransformerSaveData();
+                ctx.Writer.Write(data.Length);
+                ctx.Writer.Write(data);
+            }
+        }
+
+        public void Dispose()
+        {
+            foreach (var column in _columns)
+            {
+                column.Dispose();
+            }
+        }
+
+        #region C++ Safe handle classes
+
+        internal class TransformerEstimatorSafeHandler : SafeHandleZeroOrMinusOneIsInvalid
+        {
+            private DestroyCppTransformerEstimator _destroyCppTransformerEstimator;
+            public TransformerEstimatorSafeHandler(IntPtr handle, DestroyCppTransformerEstimator destroyCppTransformerEstimator) : base(true)
+            {
+                SetHandle(handle);
+                _destroyCppTransformerEstimator = destroyCppTransformerEstimator;
+            }
+
+            protected override bool ReleaseHandle()
+            {
+                // Not sure what to do with error stuff here. There shoudln't ever be one though.
+                return _destroyCppTransformerEstimator(handle, out IntPtr errorHandle);
+            }
+        }
+
+        internal class SaveDataSafeHandler : SafeHandleZeroOrMinusOneIsInvalid
+        {
+            private IntPtr _dataSize;
+
+            [DllImport("Featurizers", EntryPoint = "StringFeaturizer_string_DestroyTransformerSaveData", CallingConvention = CallingConvention.Cdecl), SuppressUnmanagedCodeSecurity]
+            private static extern bool DestroyTransformerSaveDataNative(IntPtr buffer, IntPtr bufferSize, out IntPtr error);
+            public SaveDataSafeHandler(IntPtr handle, IntPtr dataSize) : base(true)
+            {
+                SetHandle(handle);
+                _dataSize = dataSize;
+            }
+
+            protected override bool ReleaseHandle()
+            {
+                // Not sure what to do with error stuff here.  There shoudln't ever be one though.
+                return DestroyTransformerSaveDataNative(handle, _dataSize, out IntPtr errorHandle);
+            }
+        }
+
+        internal class TransformedDataSafeHandler : SafeHandleZeroOrMinusOneIsInvalid
```

Comment:
(minor) The .NET type is a `SafeHandle`, not `SafeHandler`. I think we should drop the `r` here and just be `TransformedDataSafeHandle`.

Reply:
Removed the extra `r` from all the `SafeHandle` classes. Fixed in next push.